llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9128,8 +9008,17 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
               : new VPInstruction(Instruction::Select, {Cond, Red, PhiR});
       Result->insertBefore(&*Builder.getInsertPoint());
       Red->replaceUsesWithIf(
-          Result->getVPSingleValue(),
-          [](VPUser &U, unsigned) { return isa<VPLiveOut>(&U); });
+          Result->getVPSingleValue(), [VectorLoopRegion](VPUser &U, unsigned) {
+            auto *UR = cast<VPRecipeBase>(&U);
+            if (UR->getParent()->getParent() != VectorLoopRegion) {
+              assert(cast<VPInstruction>(UR)->getOpcode() ==
+                         VPInstruction::ComputeReductionResult &&
+                     "user outside of the vector region must be "
+                     "ComputeReductionResult");
+              return true;
+            }
+            return false;
+          });
```

Comment:
```suggestion
          Result->getVPSingleValue(), [](VPUser &U, unsigned) {
            return isa<VPInstruction>(&U) &&
                   cast<VPInstruction>(&U)->getOpcode() == VPInstruction::ComputeReductionResult; });
```
?
It's possible for both the Select and/or ComputeReductionResult to be placed inside the loop or afterwards in the Exit block.